### PR TITLE
Update Azure.Identity to 1.11.2

### PR DESCRIPTION
**Pull request by jviau**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [ ] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Addresses https://github.com/advisories/GHSA-x674-v45j-fwxw

**ALTERNATIVE PR:** #10038

### CODE REVIEW

File: test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json [hunk 1]
```diff
@@ -893,9 +894,9 @@
       "Microsoft.Azure.Functions.JavaWorker/2.14.0": {},
       "Microsoft.Azure.Functions.NodeJsWorker/3.10.0": {},
       "Microsoft.Azure.Functions.PowerShellWorker.PS7.0/4.0.3148": {},
-      "Microsoft.Azure.Functions.PowerShellWorker.PS7.2/4.0.3131": {},
-      "Microsoft.Azure.Functions.PowerShellWorker.PS7.4/4.0.3147": {},
-      "Microsoft.Azure.Functions.PythonWorker/4.28.0": {},
+      "Microsoft.Azure.Functions.PowerShellWorker.PS7.2/4.0.3220": {},
```
Comment (kshyju) [inline test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json:897]: Looks like PR has more changes than update to Microsoft.Identity.Client package. Did you mean to include them as well?

Comment (jviau) [reply test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json:897]: Yes. These are changes that already exist in the dev branch. deps.json was just not regenerated after some worker updates.

**Review by fabiocav — COMMENTED** [review]
Can we instead bump `Microsoft.Identity.Client.Extensions.Msal` to 4.60.3? That should bring the right dependency.

**Review by fabiocav — APPROVED** [review]

File: test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json (hunk 1 above)
Comment (fabiocav) [reply test/WebJobs.Script.Tests/Microsoft.Azure.WebJobs.Script.WebHost.deps.json:897]: Those should be ignored by the validation. Good to sync periodically, but shouldn't be a blocker.

**Review by VpOfEngineering — APPROVED** [review]

**Review by kshyju — APPROVED** [review]

**Pull request merged**